packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -512,8 +512,8 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 					}
 
 					// Check config to know if execution should be saved or not
-					let saveDataErrorExecution = config.get('executions.saveDataOnError') as string;
-					let saveDataSuccessExecution = config.get('executions.saveDataOnSuccess') as string;
+					let saveDataErrorExecution = config.getEnv('executions.saveDataOnError') as string;
+					let saveDataSuccessExecution = config.getEnv('executions.saveDataOnSuccess') as string;
```

Comment:
type casting still needed?

Reply:
Removing these two assertions would require disentangling the `IWorkflowSettings` type, which would not be trivial is likely better tackled in a separate PR.